cli/pkg/kctrl/cmd/package/repository/add_or_update.go

```diff
@@ -196,7 +196,9 @@ func (o *AddOrUpdateOptions) Run(args []string) error {
 		return err
 	}
 
-	if o.URL != "" && o.URL == existingRepository.Spec.Fetch.ImgpkgBundle.Image {
+	if (o.URL != "" && o.URL == existingRepository.Spec.Fetch.ImgpkgBundle.Image) &&
```

Comment:
I think we can boil it down to `o.URL == existingRepository.Spec.Fetch.ImgpkgBundle.Image || o.SecretRef == "" || o.SecretRef == existingRepository.Spec.Fetch.ImgpkgBundle.SecretRef.Name`

Would that not cover all cases?

Reply:
I have optimised the condition as discussed